assignment2/task1.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: UTF-8 -*-
+
+
+class AncestorsFinder(object):
+    def __init__(self, graph):
+        self.graph = graph
+        self.root = 0
+        self.ancestors = None
+
+    def find(self, value):
+        self.ancestors = []
+        self._dfs(self.root, -1, value)
+        return self.ancestors
+
+    def _dfs(self, ver, prev, value):
```

Comment:
For solving this task I used  depth-first search (dfs).

Parameters:
`ver` - current vertex, that we want to check; if it contain value that we try to find, stop investigating with dfs graph and build the answer; otherwise make new dfs iteration
`prev` - previous vertex, from which we can go to current vertex; need for not checking twice one vertex
`value` - value, that we want to find in a graph